### PR TITLE
docs: Add OBS package installation guide for openSUSE

### LINKED ISSUE #160: OBS installation guide for OpenSUSE
Create an installation guide for OBS packages

### PR DESCRIPTION
**Pull request by GabyCT**
This is the installation guide for OBS packages for
openSUSE.

Fixes #160

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

**Comment by GabyCT** [comment]
Currently this is blocked by this issue https://github.com/kata-containers/packaging/issues/39

File: install/opensuse-installation-guide.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Install Kata Containers on openSUSE Leap 42.3
+
+> **Notes:**
+>
+> - Kata Containers packages are available for [openSUSE\*](https://www.opensuse.org/)
+>   (currently `x86_64` only).
```
Comment (jodh-intel) [inline install/opensuse-installation-guide.md]: For consistency with the other docs, please can you drop the "42.3" from the title (in this doc and the other one) and change this to:

> - Kata Containers packages are available for [openSUSE\*](https://www.opensuse.org/)
￼>   version 43.2 (currently `x86_64` only).

File: install/opensuse-installation-guide.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+   > **Note:** This installation channel is not secure since the repository currently
+   > redirects download URLs to `http`.
+
+   ```bash
+   $ source /etc/os-release
+   $ sudo -E VERSION_ID=$VERSION_ID zypper addrepo "https://download.opensuse.org/repositories/home:/katacontainers:/release/openSUSE_Leap_${VERSION_ID}/home:katacontainers:release.repo"
+   $ sudo -E zypper -n install kata-runtime kata-proxy kata-shim
+   ```
+
+2. Decide which container manager to use and select the corresponding link that follows::
```
Comment (jodh-intel) [inline install/opensuse-installation-guide.md]: - Can you remove the redundant 2nd colon at the end of line please?
- For consistency with the other docs can you change "to use and" to "you will use and".

File: install/docker/opensuse-docker-install.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+   $ sudo systemctl restart docker
+   ```
+
+4. Run Kata Containers
+
+   You are now ready to run Kata Containers:
+
+   ```bash
+   $ sudo docker run -ti busybox uname -a
+   ```
```
Comment (jodh-intel) [inline install/docker/opensuse-docker-install.md]: For consistency with the other docs, please add the following after this:

> The previous command shows details of the kernel version running inside the
> container, which is different to the host kernel version.

File: install/opensuse-installation-guide.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+   > **Note:** This installation channel is not secure since the repository currently
+   > redirects download URLs to `http`.
+
+   ```bash
+   $ source /etc/os-release
+   $ sudo -E VERSION_ID=$VERSION_ID zypper addrepo "https://download.opensuse.org/repositories/home:/katacontainers:/release/openSUSE_Leap_${VERSION_ID}/home:katacontainers:release.repo"
+   $ sudo -E zypper -n install kata-runtime kata-proxy kata-shim
+   ```
+
+2. Decide which container manager you will use and select the corresponding link that follows:
```
Comment (jodh-intel) [inline install/opensuse-installation-guide.md]: s/you will use/to use/ for consistency with the other guides.

File: install/docker/opensuse-docker-install.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+   $ sudo systemctl daemon-reload
+   $ sudo systemctl restart docker
+   ```
+
+4. Run Kata Containers
+
+   You are now ready to run Kata Containers:
+
+   ```bash
+   $ sudo docker run -ti busybox uname -a
```
Comment (jodh-intel) [inline install/docker/opensuse-docker-install.md]: Could you just add a blank line between the end of the code block and the sentence below it?

**Comment by GabyCT** [comment]
@jodh-intel changes applied

File: install/docker/opensuse-docker-install.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
+# Install Docker for Kata Containers on openSUSE Leap 42.3
```
Comment (jodh-intel) [inline install/docker/opensuse-docker-install.md]: Could you remove `42.3` from the end of this line please, then

lgtm

**Comment by GabyCT** [comment]
@jodh-intel changes applied thanks

**Comment by jodh-intel** [comment]
Thanks @GabyCT.

lgtm

**Comment by jodh-intel** [comment]
Ping @GabyCT.

**Comment by GabyCT** [comment]
@jodh-intel changes applied to the PR, however, the issue https://github.com/kata-containers/packaging/issues/39 has not been solved

File: install/opensuse-installation-guide.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+   ```bash
+   $ source /etc/os-release
+   $ sudo -E VERSION_ID=$VERSION_ID zypper addrepo "https://download.opensuse.org/repositories/home:/katacontainers:/release/openSUSE_Leap_${VERSION_ID}/home:katacontainers:release.repo"
+   $ sudo -E zypper -n install kata-runtime kata-proxy kata-shim
+   ```
+
+2. Decide which container manager to use and select the corresponding link that follows:
+
+   - [Docker](docker/centos-docker-install.md)
+   - [Kubernetes](https://github.com/kata-containers/documentation/blob/master/Developer-Guide.md#if-you-want-to-run-kata-containers-with-kubernetes)
```
Comment (jodh-intel) [inline install/opensuse-installation-guide.md]: This URL is now incorrect. Please can you change it to:

- https://github.com/kata-containers/documentation/blob/master/Developer-Guide.md#run-kata-containers-with-kubernetes

File: install/docker/opensuse-docker-install.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+   $ sudo systemctl daemon-reload
+   $ sudo systemctl restart docker
+   ```
+
+4. Run Kata Containers
+
+   You are now ready to run Kata Containers:
+
+   ```bash
+   $ sudo docker run -ti busybox uname -a
```
Comment (jodh-intel) [inline install/docker/opensuse-docker-install.md]: Please remove the `-ti` which we had to do for the other docker install guides to allow this line to be executed by the CI.

**Comment by GabyCT** [comment]
@jodh-intel changes applied

File: install/opensuse-installation-guide.md [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+   ```bash
+   $ source /etc/os-release
+   $ sudo -E VERSION_ID=$VERSION_ID zypper addrepo "https://download.opensuse.org/repositories/home:/katacontainers:/release/openSUSE_Leap_${VERSION_ID}/home:katacontainers:release.repo"
+   $ sudo -E zypper -n install kata-runtime kata-proxy kata-shim
+   ```
+
+2. Decide which container manager to use and select the corresponding link that follows:
+
+   - [Docker](docker/centos-docker-install.md)
```
Comment (jodh-intel) [inline install/opensuse-installation-guide.md]: Incorrect link.

**Comment by GabyCT** [comment]
@jodh-intel oops thanks changes applied

**Comment by jodh-intel** [comment]
lgtm

**Comment by jodh-intel** [comment]
This can't be merged until those labels are removed from this PR.

**Comment by GabyCT** [comment]
@jodh-intel should we merge this? even if this issue kata-containers/packaging#39 is not being solved

**Comment by jodh-intel** [comment]
Hi @GabyCT - well, could you confirm the doc can be successfully "executed" on an opensuse system to install a working Kata system? On an opensuse system with this PR branch sitting in `$GOPATH/src/github.com/kata-containers/documentation`, the following should work:

```
$ cd $GOPATH/src/github.com/kata-containers/tests/cmd/kata-manager
$ ./kata-manager.sh install-docker-system
```

**Comment by GabyCT** [comment]
@jodh-intel let me verify :)

**Comment by GabyCT** [comment]
@jodh-intel , I check that
```
$ ./kata-manager.sh install-docker-system
```
and it is working

File: install/opensuse-installation-guide.md [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+>   [`kata-doc-to-script`](https://github.com/kata-containers/tests/blob/master/.ci/kata-doc-to-script.sh) script.
+
+1. Install the Kata Containers components with the following commands:
+
+   > **Note:** This installation channel is not secure since the repository currently
+   > redirects download URLs to `http`.
+
+   ```bash
+   $ source /etc/os-release
+   $ sudo -E VERSION_ID=$VERSION_ID zypper addrepo "https://download.opensuse.org/repositories/home:/katacontainers:/release/openSUSE_Leap_${VERSION_ID}/home:katacontainers:release.repo"
```
Comment (jodh-intel) [inline install/opensuse-installation-guide.md]: Until OBS/Suse resolve the https -> http issue for OpenSuSE+SLES (https://github.com/kata-containers/documentation/pull/83#issuecomment-412438187), this URL needs to be http imho.

**Comment by jodh-intel** [comment]
\o/ :smile:

Once you've tweaked that url from https to http...

lgtm

**Comment by GabyCT** [comment]
@jodh-intel changes applied

**Comment by jodh-intel** [comment]
Thanks @GabyCT.

lgtm

**Comment by amshinde** [comment]
@egernst PTAL

**Comment by jodh-intel** [comment]
Ping @egernst, @grahamwhaley, @bergwolf.

File: install/opensuse-installation-guide.md [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+
+   > **Note:** This installation channel is not secure since the repository currently
+   > redirects download URLs to `http`.
+
+   ```bash
+   $ source /etc/os-release
+   $ sudo -E VERSION_ID=$VERSION_ID zypper addrepo "http://download.opensuse.org/repositories/home:/katacontainers:/release/openSUSE_Leap_${VERSION_ID}/home:katacontainers:release.repo"
+   $ sudo -E zypper -n --no-gpg-checks install kata-runtime kata-proxy kata-shim
+   $ sudo mkdir -p /usr/libexec/kata-containers
+   $ sudo ln -s /usr/lib64/kata-containers/kata-proxy /usr/libexec/kata-containers/kata-proxy
```
Comment (bergwolf) [inline install/opensuse-installation-guide.md]: Any reason `kata-proxy` and `kata-shim` are not installed under `/usr/libexec/kata-containers` in the first place?

Comment (jodh-intel) [reply install/opensuse-installation-guide.md]: Hi @bergwolf - it's due to https://github.com/kata-containers/packaging/issues/39. We need someone with access to a suse system and packaging skills to resolve the issue with the spec file.

**Comment by jcvenegas** [comment]
@bergwolf  @GabyCT @jodh-intel the /usr/libexec  issue should be fixed now give it a try again.

**Comment by jodh-intel** [comment]
Adding dnm label so @GabyCT can test. If it works, we can drop the workaround in this doc...

**Comment by GabyCT** [comment]
@jcvenegas and @jodh-intel , this was tested in the `alpha` repository and is working, the workaround has been removed

**Comment by amshinde** [comment]
lgtm

File: install/docker/opensuse-docker-install.md [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+
+   ```bash
+   $ sudo zypper -n install libcgroup1
+   $ sudo zypper -n install docker
+   ```
+
+   For more information on installing Docker please refer to the
+   [Docker Guide](https://software.opensuse.org/package/docker).
+
+2. Configure Docker to use Kata Containers with one of the following methods:
```
Comment (jodh-intel) [inline install/docker/opensuse-docker-install.md]: For consistency with the other guides, please change this to the following (which includes "by default"):

> 2. Configure Docker to use Kata Containers by default with one of the following methods:

**Comment by jodh-intel** [comment]
Once the "by default" wording has been added,

lgtm

**Comment by GabyCT** [comment]
@jodh-intel done

**Pull request merged**